content/how-to-use-supertokens-custom-ui-with-vuejs/index.md

```diff
@@ -0,0 +1,608 @@
+              return;
+          }
+
+          if (response.status === "FIELD_ERROR") {
+              response.formFields.forEach((item) => {
+                  if (item.id === "email") {
+                      // this means that something was wrong with the entered email.
+                      // probably that it's not a valid email (from a syntax point of view)
+                      this.emailError = item.error;
+                  } else if (item.id === "password") {
+                      this.passwordError = item.error;
+                  }
+              });
+              return;
+          }
+
+          // login is successful, and we redirect the user to the home page.
+          // Note that session cookies are added automatically and nothing needs to be
+          // done here about them.
+          window.location.assign("/");
+      },
+
+```
+
+In case of `FIELD_ERROR`, it means that the user entered an invalid email so we store the error state and display the error message on the UI to the user. Here’s how the error message appears underneath the email field:
+
+![invalid-email-error](./invalid-email-error.png)
+
+Similarly, we have the `signUp` method where we invoke the `emailPasswordSignUp` function from `supertokens-web-js` to handle the signup flow. 
+
+For social login, we’re using Google and Github Authentication providers. When the `onGithubPressed` or `onGooglePressed` functions are called, we call the `getAuthorisationURLWithQueryParamsAndSetState` method and pass the provider name in the parameters. We also provide a callback URL as `authorisationURL` parameter that each provider will redirect to after the authentication process is completed. In our example, we use http://localhost:3000/auth/callback/google for Google and [http://localhost:3000/auth/callback/github for GitHub . Note that this URL needs to be configured on the provider’s dashbaord.
+
+Here are the functions for Github and Google respectively:
+
+```vue
+...
+      onGithubPressed: async function () {
+          const authUrl = await ThirdPartyEmailPassword.getAuthorisationURLWithQueryParamsAndSetState({
+              providerId: "github",
+
+              // This is where github should redirect the user back after login or error.
+              // This URL goes on the github dashboard as well.
+              authorisationURL: `http://localhost:3000/auth/callback/github`,
+          });
+
+          window.location.assign(authUrl);
+      },
+      onGooglePressed: async function () {
+          const authUrl = await ThirdPartyEmailPassword.getAuthorisationURLWithQueryParamsAndSetState({
+              providerId: "google",
+
+              // This is where google should redirect the user back after login or error.
+              // This URL goes on the google dashboard as well.
+              authorisationURL: `${websiteDomain}/auth/callback/google`,
+          });
+
+          window.location.assign(authUrl);
+      },
+ ……
+```
```

Comment:
```suggestion
```ts
onGithubPressed: async function () {
    const authUrl = await ThirdPartyEmailPassword.getAuthorisationURLWithQueryParamsAndSetState({
        providerId: "github",

        // This is where github should redirect the user back after login or error.
        // This URL goes on the github dashboard as well.
        authorisationURL: "http://localhost:3000/auth/callback/github",
    });

    window.location.assign(authUrl);
},

onGooglePressed: async function () {
    const authUrl = await ThirdPartyEmailPassword.getAuthorisationURLWithQueryParamsAndSetState({
        providerId: "google",

        // This is where google should redirect the user back after login or error.
        // This URL goes on the google dashboard as well.
        authorisationURL: "http://localhost:3000/auth/callback/google",
    });

    window.location.assign(authUrl);
}
```
```